flow/connectors/mysql/mysql.go

```diff
@@ -22,10 +23,11 @@ import (
 
 type MySqlConnector struct {
 	*metadataStore.PostgresMetadata
-	config *protos.MySqlConfig
-	ssh    utils.SSHTunnel
-	conn   *client.Conn
-	logger log.Logger
+	config   *protos.MySqlConfig
+	ssh      utils.SSHTunnel
+	conn     atomic.Pointer[client.Conn]
```

Comment:
note that the use of atomics here does not expose a threadsafe interface, this is only to avoid race conditions on `c.conn` being replaced during retries & context watcher reading `c.conn`

Reply:
can be a comment just in case